ts/modules/pattern-fill.src.ts

```diff
@@ -740,13 +740,11 @@ addEvent(H.Chart, 'redraw', function (): void {
                 var id = node.getAttribute('fill') ||
                         node.getAttribute('color') ||
                         node.getAttribute('stroke');
-
                 if (id) {
-                    usedIds.push(
-                        id
-                            .substring(id.indexOf('url(') + 5)
-                            .replace(')', '')
-                    );
+                    const sanitizedId = id.replace(renderer.url, '').replace('url(#', '').replace(')', '');
+                    if (usedIds.indexOf(sanitizedId) === -1) {
```

Comment:
I see that we are using an array to keep track of the used IDs. We should probably just use a hash-type lookup object for this, e.g. `usedIds[sanitizedId] = true` where `usedIds = {}`. Then below where we loop through we can just run `usedIds[patternId]` to test if it exists - this should be much faster than searching the array.

Reply:
Like so?

Reply:
Yep, (assuming this works)!